### PR TITLE
feat:  Setting of Routes Skeletons[REST API](#39523)

### PR DESCRIPTION
**Pull request by nxfareed**
Link for todays task: https://redmine.successive.tech/issues/39523

### CODE REVIEW

File: src/controllers/trainee/routes.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Router } from 'express';
+import traineeController from './controller';
+
+const traineeRouter = Router();
+
+traineeRouter.route('/trainee')
+.get( traineeController.create )
+.post( traineeController.list)
+.put( traineeController.update)
+.delete( traineeController.delete);
```
Comment (Prabal555) [inline src/controllers/trainee/routes.ts]: destructure the values at the top

File: src/controllers/trainee/controller.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import { Request, Response } from 'express';
```
Comment (Prabal555) [inline src/controllers/trainee/controller.ts:1]: Indentation in the file

File: src/controllers/trainee/routes.ts (hunk 1 above)
Comment (Prabal555) [reply src/controllers/trainee/routes.ts]: indentation is also not correct

File: tsconfig.json [hunk 3]
```diff
@@ -1,4 +1,33 @@
...
+"exclude": ["dist"],
+"include": [
+"src"
+],
+"types": ["src/types"]
+=======
+>>>>>>> 4989eedde3b667c7e15101ae4915eb03cb683244
     "compilerOptions": {
         "module": "commonjs",
         "target": "es6",
```
Comment (Prabal555) [inline tsconfig.json]: conflicts have been pushed

File: tsconfig.json [hunk 4]
```diff
@@ -1,4 +1,33 @@
...
+"src/types/*"
+]
+},
+},
+"exclude": ["dist"],
+"include": [
+"src"
+],
+"types": ["src/types"]
+=======
```
Comment (Prabal555) [inline tsconfig.json]: Indentation is missing

**Review by Prabal555 — CHANGES_REQUESTED** [review]

File: src/Server.ts [hunk 5]
```diff
@@ -30,8 +31,18 @@ class Server {
...
+        app.use('/body-parser', (req: Request, res, next) => {
+            console.log('Inside Middleware');
+            console.log(req.body);
+            res.send(req.body);
+            });
+            
+            app.use('/api', mainRouter);
+            
+            app.use(notFoundRoutes);
+            app.use(errorHandler);
```
Comment (Prabal555) [inline src/Server.ts]: indentation is not correct

File: src/controllers/trainee/validation.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
+export const validation = {
+    create:
+    {
```
Comment (Prabal555) [inline src/controllers/trainee/validation.ts:3]: indentation

**Review by Prabal555 — CHANGES_REQUESTED** [review]

File: src/controllers/trainee/validation.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        id: {
+            required: true,
+            string: true,
+            in: ['body'],
+            custom: ((value) => {
+                console.log('now you are in custom');
+                if (!value) { }
+            })
+        },
```
Comment (Prabal555) [inline src/controllers/trainee/validation.ts:12]: not complete

File: src/controllers/trainee/validation.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        dataToUpdate: {
+            in: ['body'],
+            required: true,
+            isObject: true,
+            custom: (dataToUpdate) => {
+                {
+                    console.log('now you are in custom');
+                    if (!dataToUpdate) { };
+                }
+            },
```
Comment (Prabal555) [inline src/controllers/trainee/validation.ts:58]: code incomplete

**Review by Prabal555 — CHANGES_REQUESTED** [review]

File: src/controllers/trainee/validation.ts (hunk 8 above)
Comment (nxfareed) [reply src/controllers/trainee/validation.ts:58]: This has been left blank for the time when we have to add a  particular custom operation when required.

**Pull request merged**